fili-core/src/main/java/com/yahoo/bard/webservice/table/PhysicalTable.java

```diff
@@ -137,14 +140,19 @@ public synchronized void resetColumns(SegmentMetadata segmentMetadata, Dimension
             workingIntervals.clear();
             for (Map.Entry<String, Set<Interval>> nameIntervals : dimensionIntervals.entrySet()) {
                 String physicalName = nameIntervals.getKey();
-                String apiName = getLogicalColumnName(physicalName);
-                Dimension dimension = dimensionDictionary.findByApiName(apiName);
-                // Schema evolution may lead to unknown dimensions, skip these
-                if (dimension == null) {
-                    continue;
-                }
-                DimensionColumn dimensionColumn = DimensionColumn.addNewDimensionColumn(this, dimension);
-                workingIntervals.put(dimensionColumn, nameIntervals.getValue());
+
+                getLogicalColumnNames(physicalName).stream()
+                        .map(dimensionDictionary::findByApiName)
+                        .filter(dimension -> dimension != null)
+                        .forEach(
+                                dimension -> {
+                                    DimensionColumn dimensionColumn = DimensionColumn.addNewDimensionColumn(
```

Comment:
I'm with @garyluoex on this one. It's not a big deal, especially since we don't run around randomly making streams parallel (and we shouldn't, not without carefully examining the operations in the stream first), but it just feels deeply and fundamentally __wrong__ to me to put a side-effectful operation in a `map`.